src/sdltiles.cpp

```diff
@@ -2029,6 +2067,37 @@ void load_soundset() {
             playlists[playlist_id] = playlist_to_load;
         }
     }
+
+    // Load sound effects. This loads the sound effect chunks directly
+    // into memory.
+    location = FILENAMES["datadir"] + "/sound/soundset.json";
+    std::ifstream jsonstream2(location.c_str(), std::ifstream::binary);
+    if (jsonstream2.good()) {
+        JsonIn json(jsonstream2);
+        JsonObject config = json.get_object();
+        JsonArray sound_effects_json = config.get_array("sound_effects");
+        for (unsigned i=0; i < sound_effects_json.size(); i++) {
+            JsonObject sound_effect = sound_effects_json.get_object(i);
+
+            std::string sound_effect_id = sound_effect.get_string("id");
+            std::string sound_effect_variant = sound_effect.get_string("variant", "default");
+            sound_effect_entry sfx_to_load;
+
+            const std::string filename = sound_effect.get_string("file");
+            const std::string path = (FILENAMES["datadir"] + "/sound/" + filename);
+            Mix_Chunk *loaded_chunk = Mix_LoadWAV(path.c_str());
```

Comment:
If you go with the `shared_ptr`, you should clear the map before calling `Mix_CloseAudio` in `WinDestroy`. Clearing the map will delete all the chunks and should probably be done before the audio subsystem is closed.